File: frontend/src/api/expenses.ts
import { apiClient } from "./client"
import type { ExpenseCreate, ExpenseResponse } from "./types"

export const getExpenses = async () => {
  const response = await apiClient.get<ExpenseResponse[]>("/api/expenses")
  return response.data
}

export const createExpense = async (data: ExpenseCreate) => {
  const response = await apiClient.post<ExpenseResponse>("/api/expenses", data)
  return response.data
}

File: backend/routers/expenses.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List
from database import SessionLocal
from dependencies import get_db, get_current_user
from models.expense import Expense
from models.user import User
from schemas.expense import ExpenseCreate, ExpenseResponse

router = APIRouter(prefix="/api/expenses", tags=["expenses"])

@router.get("", response_model=List[ExpenseResponse])
def get_expenses(db: Session = Depends(get_db), current_user: User = Depends(get_current_user)):
    return db.query(Expense).all()

@router.post("", response_model=ExpenseResponse)
def create_expense(expense: ExpenseCreate, db: Session = Depends(get_db), current_user: User = Depends(get_current_user)):
    new_expense = Expense(**expense.model_dump())
    db.add(new_expense)
    db.commit()
    db.refresh(new_expense)
    return new_expense
